File: Week1/roman_to_int.java
package Week1;

import java.util.*;

public class roman_to_int {
    public static int romanToInt(String s) {
        HashMap<Character, Integer> romanValues = new HashMap<>();
        romanValues.put('I', 1);
        romanValues.put('V', 5);
        romanValues.put('X', 10);
        romanValues.put('L', 50);
        romanValues.put('C', 100);
        romanValues.put('D', 500);
        romanValues.put('M', 1000);
        
        int total = 0;
        int prevValue = 0;
        
        for (int i = 0; i < s.length(); i++) {
            char currentChar = s.charAt(i);
            int value = romanValues.get(currentChar);
            
            if (value > prevValue && prevValue != 0) {
                total += value - 2 * prevValue;
            } else {
                total += value;
            }
            
            prevValue = value;
        }
        
        return total;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a Roman numeral: ");
        String romanNumeral = scanner.nextLine().toUpperCase();
        scanner.close();

        int result = romanToInt(romanNumeral);
        if (result != -1) {
            System.out.println("The integer value of " + romanNumeral + " is: " + result);
        } else {
            System.out.println("Invalid Roman numeral entered!");
        }
    }
}

File: Week1/roman_to_int.py
def roman_to_int(s: str) -> int:
    roman_values = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    }
    total = 0
    prev_value = 0
    for char in s:
        value = roman_values.get(char)
        if value is None:
            return -1  # Invalid Roman numeral
        if value > prev_value:
            total += value - 2 * prev_value
        else:
            total += value
        prev_value = value
    return total

# Test the function
roman_numeral = input("Enter a Roman numeral: ").upper()
result = roman_to_int(roman_numeral)
if result != -1:
    print(f"The integer value of {roman_numeral} is: {result}")
else:
    print("Invalid Roman numeral entered!")
